sklearn/utils/_testing.py

```diff
@@ -188,14 +188,18 @@ def assert_warns_message(warning_class, message, func, *args, **kw):
         # Checks the message of all warnings belong to warning_class
         for index in [i for i, x in enumerate(found) if x]:
             # substring will match, the entire message with typo won't
-            msg = w[index].message  # For Python 3 compatibility
+            msg, mtb = w[index].message, ''  # For Python 3 compatibility
```

Comment:
Please explain why this change is necessary

Reply:
For ease of backward and forward compatibility, it seemed ideal to check with the original format without the stack-trace, then have this mtb variable (message without trace back) where multiline messages have their first and last line taken and spliced together for simple test comparison in case the original one fails which it will when a stack trace is in the center.

Basically, now we can compare specific stack traces too, for if we always take the first and last line for comparison which I had done earlier, then we could never compare the stack trace.  Yet the unit test required comparing the stack trace and so it felt we should not always remove it.  So now my new unit test can work with this functionality, along with all the old ones, the original check now checking with the stacktrace, and the fallback check checking with it removed.